### PR TITLE
Destroy all profiles in a policy on policy destroy

### PR DESCRIPTION
**Pull request by akofink**
Signed-off-by: Andrew Kofink <akofink@redhat.com>

### CODE REVIEW

**Comment by akofink** [comment]
I'm working on the rubocop failure. :see_no_evil:

File: app/models/concerns/profile_policy_association.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Methods that are related to a profile's policy
+module ProfilePolicyAssociation
```
Comment (akofink) [inline app/models/concerns/profile_policy_association.rb:4]: Rubocop was complaining about the length of `profile.rb`

**Comment by akofink** [comment]
This one should be good for review.

File: app/models/concerns/profile_policy_association.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    def business_objective_id
+      (policy || self).read_attribute(:business_objective_id)
+    end
+
+    def compliance_threshold
+      (policy || self).read_attribute(:compliance_threshold)
+    end
+
+    def destroy_policy_test_results
+      policy_profiles.destroy_all
```
Comment (dLobatog) [inline app/models/concerns/profile_policy_association.rb:37]: This is potentially OOM, if it has to destroy a couple of 1000s test results, can we do this asynchronously? Something like `app/jobs/delete_host.rb` would work

**Comment by akofink** [comment]
Updated to destroy all other profiles in this policy asynchronously. The first profile, though, is still destroy synchronously so the UI updates. Let me know if it's good enough.

File: app/models/concerns/profile_policy_association.rb (hunk 1 above)
Comment (dLobatog) [reply app/models/concerns/profile_policy_association.rb:4]: :+1: it's definitely tidier

**Pull request merged**